userspace/libsinsp/sinsp.cpp

```diff
@@ -2243,6 +2243,40 @@ void sinsp::init_k8s_client(string* api_server, string* ssl_cert, string* node_n
 	}
 }
 
+void sinsp::validate_k8s_node_name()
+{
+
+	if(!m_k8s_node_name || m_k8s_node_name->size() == 0)
+	{
+		m_k8s_node_name_validation_state = NO_NODE_NAME_FILTER;
+	}
+	else 
+	{
+		bool found = false;
+		const auto& state = m_k8s_client->get_state();
+
+		for(const auto& node : state.get_nodes())
+		{
+			if(!node.get_node_name().compare(*m_k8s_node_name))
+			{
+				found = true;
+				break;
+			} 
+		}
+
+		if(!found)
+		{
+			m_k8s_node_name_validation_state = INVALID_NODE_NAME_FILTER;
+		}
+		else 
+		{
+			m_k8s_node_name_validation_state = VALID_NODE_NAME_FILTER;
```

Comment:
Is m_k8s_node_name_validation_state used elsewhere? I cannot find anything else about it!

Reply:
That is meant to be used by libs consumers! Check also [this one](https://github.com/falcosecurity/falco/pull/1823) for an example. Please let me know what do you think about this!

Reply:
Uh i see! I though this was self contained inside libs :) 
Thank you!